### DISCUSSION TITLE
Hybrid Property with Relationships

### DISCUSSION THREAD

**Discussion by heavenlygaze** [sqlalchemy/sqlalchemy#10799] [Usage Questions]
I am not sure whether this question related to SQLAlchemy or Mypy, so please correct me If I am wrong for posting it here.

I was reading the [documentation](https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html) about implementing hybrid property attributes of my model, and when I run mypy on my app it was throwing a redefinition error, since I the name of the method for my hybrid property and the expression is same

**Comment by glatterf42** [comment]
If you upgraded to >v2.0 of sqlalchemy already, you can benefit from the [`inplace` marker](https://docs.sqlalchemy.org/en/20/orm/extensions/hybrid.html#using-inplace-to-create-pep-484-compliant-hybrid-properties). Something similar might exist before v2.0, too.
The idea is that by using `@property.inplace.expression`, you can use any function name you want (to satisfy mypy by avoiding a redefinition) while sqlalchemy still knows that this function is what should be substituted as the SQL expression for your `property`.
